drivers/volume/portworx/portworx.go

```diff
@@ -1212,41 +1212,42 @@ func (p *portworx) checkAndUpdateHaLevel(volID, uid string, params map[string]st
 		return false, nil
 	}
 	logrus.Infof("Restore Volume: %v \t Status %v \t state %v", restoreVol.Id, restoreVol.Status, restoreVol.State)
-	// find replica nodes
+	// find replica pools ids
 	var pPools []string
 	var rPools []string
-	var pools []string
-	// get all parent volume replica nodes
+	var replPool string
+	// get all parent volume replica pools
 	for _, rs := range parentVols[0].GetReplicaSets() {
-		pPools = append(pPools, rs.GetNodes()...)
+		pPools = append(pPools, rs.GetPoolUuids()...)
 	}
 	// get all restore volume replica nodes
 	for _, rs := range restoreVol.GetReplicaSets() {
-		rPools = append(rPools, rs.GetNodes()...)
+		rPools = append(rPools, rs.GetPoolUuids()...)
 	}
 
-	// parentvolume replica nodes - restorevolume replica nodes
-	for _, pnode := range pPools {
+	logrus.Debugf("parent volume pools %v", pPools)
+	logrus.Debugf("restore volume pools %v", rPools)
+	// get pool to perform ha update
+	for _, pool := range pPools {
 		isPart := false
-		for _, rnode := range rPools {
+		for _, rpool := range rPools {
 			// since restore volumes rs will always be 1
-			if pnode == rnode {
-				logrus.Debugf("skipping node %v", rnode)
+			if pool == rpool {
 				isPart = true
 				break
 			}
 		}
-		logrus.Debugf("adding node %v", pnode)
 		if !isPart {
-			pools = append(pools, pnode)
+			replPool = pool
+			break
 		}
 	}
 
 	if restoreVol.GetSpec().GetHaLevel() != parentVols[0].GetSpec().GetHaLevel() {
-		logrus.Infof("Updating HA Level of volume %v", restoreVol.GetLocator().GetName())
+		logrus.Infof("Updating HA Level of volume %v to pool %v", restoreVol.GetLocator().GetName(), replPool)
 		spec := &api.VolumeSpec{
 			HaLevel:    restoreVol.GetSpec().GetHaLevel() + 1,
```

Comment:
If HA level of parent of volume is 3, would updating HA level to HA level + 1  work ? (it would make it 2 not 3)

Reply:
`checkAndUpdateHALevel` is called till ha level of restore volume does not match parent volume. At a time you can only update ha level by 1 of px volume

https://github.com/libopenstorage/stork/pull/514/files/9af000cba70565967f4e0458a6b257d2335c422e#diff-0aa9aa57d00bdd00603aae645329fb04R1176

Reply:
https://github.com/libopenstorage/stork/blob/9af000cba70565967f4e0458a6b257d2335c422e/drivers/volume/portworx/portworx.go#L1155